File: sliding window/LengthOfLongestSubstring.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        start_window = 0
        max_length = 0
        dict = {}
        
        for i in range(len(s)):
            ch = s[i]
            if ch in dict:
                start_window = max(start_window, dict[ch] + 1)
            dict[ch] = i
            max_length = max (max_length, i-start_window+1)
        return max_length

File: sliding window/LengthOfLongestSubstring.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        int max_length = 0;
        int start_window = 0;
        Dictionary<char, int> dict = new Dictionary<char, int>();
        
        for (int i=0; i<s.Length; ++i){          
            char currentChar = s[i];
            if(dict.ContainsKey(currentChar))
                start_window = Math.Max(start_window, dict[currentChar] + 1);
            dict[currentChar] = i;
            max_length = Math.Max(max_length, i - start_window + 1);
        }
        return max_length;
    }
}